### PR TITLE
Added some tips for reviewers to the PR docs

### PR DESCRIPTION
**Pull request by beav**
As a new contributor, I had some trouble finding PRs ready for review. I
created a small github query to assist with this.

### CODE REVIEW

File: developers/pull_request_checklist.md [hunk 1]
```diff
@@ -46,3 +46,14 @@ another developer.
 * Are there any comments that haven't been addressed?
 * Did someone "ACK" the PR?
 * Is the PR still green?
+
+# Tips for reviewers
```
Comment (ehelms) [inline developers/pull_request_checklist.md]: The text is very new developer focused so many this belongs more so in the becoming a contributor section? As we chatted in IRC, I don't necessarily agree with some of the limitations of this (PRs can be reviewed before tests pass, sometimes design changes things and would make your tests invalid, and reviewing larger PRs that include design changes is something I would encourage -- they may be daunting but you can learn a lot).

**Comment by beav** [comment]
This fails because the newly added github URL will 404 unless you add `Accept: text/html` in the header. I'll see if there is a workaround.

**Comment by stbenjam** [comment]
@beav You can include the query link if you do an HTML link and add  data-proofer-ignore to it: https://github.com/gjtorikian/html-proofer#ignoring-content

**Comment by beav** [comment]
(going through some older PRs to clear out)

@stbenjam  i do not remember my previous query, I have not been using the query much anyway since I wrote this so it is probably not worth documenting :koala: 

Is there any additional comment?

**Comment by stbenjam** [comment]
I don't remember either, but this looks good to me, ACK

**Pull request merged**